charity_finder/views.py

```diff
@@ -1,14 +1,61 @@
+import folium
 from django.shortcuts import render, get_object_or_404
 from django.http import HttpResponse, JsonResponse
 from pprint import pprint
 from functools import partial
+from folium.plugins import HeatMap
 
 from charity_finder.models import Theme, Organization, Project
 from charity_finder import charity_api
 
 # Create your views here.
 def home(request):
-    return render(request, "home.html")
+
+    m = folium.Map(location=[59.09827437369457, 13.115860356662202], zoom_start=3)
+
+    # TODO: have smaller goals but then limit it to continent or country
+    projects = Project.objects.filter(goal_remaining__gte=500_000).values(
+        "title", "project_link", "latitude", "longitude", "goal_remaining"
+    )
+
+    for project in projects:
+        if project["latitude"] and project["longitude"] and project["goal_remaining"]:
```

Comment:
@bbelderbos So would this look something like this?? I tried this but VSCode wasn't able to recognize the attribute in views.py.  I wonder how this works if I am doing a for loop on the model instance.  Sources: https://dev.to/thepylot/using-property-decorator-in-django-p0a, https://www.andreadiotallevi.com/blog/how-to-use-the-property-decorator-in-python-and-django. (Noting the Django Limitations of properties with QuerySets).  

```
# usage in models.py
@property
def has_map_data(self):
    if self.latitude and self.longitude and self.goal_remaining:
        return True

# usage in views.py
for project in projects:
    if project.has_map_data:
    # do stuff

Reply:
Yes if added `has_map_data` to `Project` that should work. You can shorten it to:

```
@property
def has_map_data(self):
    return self.latitude and self.longitude and self.goal_remaining
```

Which has the additional advantage to always return a boolean, in your version in case of the else it would return None (Python's default return object)

Reply:
Actually this was not always returning bool, because it returns the first "truthy" variable which might not be True, wrapping it in bool() (+ type hint = nice) we always get a boolean:

```
    @property
    def has_map_data(self) -> bool:
        return bool(self.latitude and self.longitude and self.goal_remaining)
```